keps/sig-storage/0661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,808 @@
+performed to the PVC object.
+
+### Revision control
+
+StatefulSet supports [revision control](https://kubernetes.io/blog/2017/09/kubernetes-statefulsets-daemonsets/).
+The storage size change will not be added to the fields supporting revision control. Any change in
+revision - going forward to a new version or going back to an older version would not make any
+changes to the persistent volume size. Any changes requiring resize of the volume will need editing
+of the StatefulSet storage size specification.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+Controller revision is a very useful functionality. As specified earlier in this doc, no changes will be made to the
+Statefulset controller revision code. This will facilitate cases where statefulset has to roll back to a previous 
+version with smaller requested storage size without the need to shrink the volume. Detailed documentation will be 
+added to mitigate any user expectation mismatch.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### API server validation relaxation
+Modifications would be made to `ValidateStatefulSetUpdate` in order to allow changes in the
+storage size of the volume in the `VolumeClaimTemplates`. The new requested size will be validated to 
+prevent a user from shrinking the volume.
+
+### StatefulSet controller changes
+The `sync` function in the `StatefulSetController` runs the reconciliation to match the current
+reality and the expected declaration in the `Spec`. An addendum will be made to validate each
+PVC associated with each pod and ensure that their sizes are matching VolumeClaimTemplate specification.
+If not, a call is made to update the PVC object with the size specified in the VolumeClaimTemplate.
```

Comment:
This should probably be,
If the pvc's size is less then the size specified in the statefulset.spec, it will be updated.

If the user manually updated the pvc to a bigger value, this behavior will prevent an attempt to srhink.

Reply:
Updated it. Thank you!